### DISCUSSION TITLE
What is the error about maximum snapshot size about

### DISCUSSION THREAD

**Discussion by tcoopman** [jj-vcs/jj#2195] [Q&A] [answered]
I've just had this error when adding a new image file:
`Internal error: Failed to snapshot the working copy: New file .../priv/static/images/postits-pile.png
of size ~1.5MiB exceeds snapshot.max-new-file-size (1.0MiB)`

I've found out how to fix it (setting a bigger snapshot.max-new-file-size), but I'm not sure what this setting is about. I've found the PR that merged this in, but there is no reason given why this setting is added or if there are any issues with setting a bigger value.
Can someone give me some context about this?

**Comment by martinvonz** [comment, marked as answer]
The idea is to prevent accidentally committing large files and increasing the size of your repo. If you're not worried about that, there's no reason not to increase the limit. It seems like we can at least improve the error message.

**Comment by martinvonz** [comment]
I created #2198 to make the error a little clearer.
